test/cli/test-other.py

```diff
@@ -825,3 +826,41 @@ def test_file_duplicate_2(tmpdir):
         '3/3 files checked 0% done'
     ]
     assert stderr == ''
+
+
+def test_premium_with_relative_path(tmpdir): #12254
+
+
+    test_file = os.path.join(tmpdir, 'test.c')
+    with open(test_file, 'wt'):
+        pass
+
+    product_name = 'Cppcheck Premium ' + str(time.time())
+
+    test_cfg = lookup_cppcheck_exe()
+    if(test_cfg.endswith('.exe')):
```

Comment:
This is very bad and dangerous. If the test gets aborted beforehand you have a stray file which will infer with your system. It might also override a file the user might have created for themselves.
A better way to test this would probably be to copy the folder where the binary is located in into a new folder within `tmpdir`.

Reply:
We already do something similar in other tests with the local `cppcheck_local()` - which is not something I would encourage. I will try to provide a shared helper which will generate a copy for the execution and update the existing tests.